src/staticanalysis/bot/static_analysis_bot/workflow.py

```diff
@@ -275,6 +272,14 @@ def detect_issues(self, analyzers, revision):
             # Run analyzer on revision and store generated issues
             issues += analyzer.run(revision)
 
+            # Clean up any uncommitted changes left behind by this analyzer.
+            self.hg.revert(settings.repo_dir.encode('utf-8'), all=True)
+            logger.info('Reverted all uncommitted changes in repo', rev=self.hg.identify())
+
+            # Compute line hashes now before anything else changes the code.
+            for issue in issues:
+                issue.build_lines_hash()
```

Comment:
You are re-building lines hash on all issues here, not just the new ones.

Reply:
I don't think any issue hash has been calculated at that point.

Reply:
Oh wow, that's true. Thanks for the nice catch!

Reply:
Fixed like so:

```diff
diff --git a/src/staticanalysis/bot/static_analysis_bot/workflow.py b/src/staticanalysis/bot/static_analysis_bot/workflow.py
index 35d1a1c1..c9c81854 100644
--- a/src/staticanalysis/bot/static_analysis_bot/workflow.py
+++ b/src/staticanalysis/bot/static_analysis_bot/workflow.py
@@ -270,16 +270,18 @@ class Workflow(object):
             analyzer = analyzer_class()
 
             # Run analyzer on revision and store generated issues
-            issues += analyzer.run(revision)
+            analyzer_issues += analyzer.run(revision)
 
             # Clean up any uncommitted changes left behind by this analyzer.
             self.hg.revert(settings.repo_dir.encode('utf-8'), all=True)
             logger.info('Reverted all uncommitted changes in repo', rev=self.hg.identify())
 
             # Compute line hashes now before anything else changes the code.
-            for issue in issues:
+            for issue in analyzer_issues:
                 issue.build_lines_hash()
 
+            issues += analyzer_issues
+
         return issues
 
     def index(self, revision, **kwargs):
```